src/daemon/daemon.cpp

```diff
@@ -135,7 +135,7 @@ bool t_daemon::run(bool interactive)
 
     mp_internals->p2p.run(); // blocks until p2p goes down
 
-    if (interactive)
+    if (rpc_commands)
```

Comment:
Oops, I was looking at the updated check, not the original. Definitely need a `null` assignment now.